pandas/tests/frame/methods/test_update.py

```diff
@@ -146,3 +148,43 @@ def test_update_with_different_dtype(self):
 
         expected = DataFrame({"a": [1, 3], "b": [np.nan, 2], "c": ["foo", np.nan]})
         tm.assert_frame_equal(df, expected)
+
+    @given(dtype=np_st.integer_dtypes())
```

Comment:
don't use hypthoesis, its if for very specific tests. we *already* have a fixture for this. any_nullable_dtype (there are many fixtures if you want to comprehensively test this in pandas/conftest.py)

Reply:
Alright! Thanks for the hint, I'll look it up.